ray-operator/controllers/ray/common/pod.go

```diff
@@ -243,34 +243,80 @@ func DefaultWorkerPodTemplate(instance rayv1.RayCluster, workerSpec rayv1.Worker
 	return podTemplate
 }
 
-// For KubeRay, the liveness and readiness probes perform the same checks.
```

Comment:
Why do we need to split this function into two? The two functions share much of the same code logic.